server/server.go

```diff
@@ -114,13 +114,15 @@ var (
 )
 
 const (
-	firebaseControlTopic     = "~control"                // See Android if changed
-	firebasePollTopic        = "~poll"                   // See iOS if changed
-	emptyMessageBody         = "triggered"               // Used if message body is empty
-	newMessageBody           = "New message"             // Used in poll requests as generic message
-	defaultAttachmentMessage = "You received a file: %s" // Used if message body is empty, and there is an attachment
-	encodingBase64           = "base64"                  // Used mainly for binary UnifiedPush messages
-	jsonBodyBytesLimit       = 16384
+	firebaseControlTopic        = "~control"                // See Android if changed
+	firebasePollTopic           = "~poll"                   // See iOS if changed
+	emptyMessageBody            = "triggered"               // Used if message body is empty
+	newMessageBody              = "New message"             // Used in poll requests as generic message
+	defaultAttachmentMessage    = "You received a file: %s" // Used if message body is empty, and there is an attachment
+	encodingBase64              = "base64"                  // Used mainly for binary UnifiedPush messages
+	jsonBodyBytesLimit          = 16384
+	subscriberBilledTopicPrefix = "up_"
```

Comment:
1. We have to find a better word for "billing". It is confusing with all the payment work in the codebase. I do like the term "subscriberBilling", but I think the more semantically correct term would be "subscriberBasedRateLimiting" (albeit a little long)

2. I love your example of how it's useful elsewhere. And it's true, I can see that. Uhhh, what if it's not based on a prefix at all, but on a header that the subscriber sends, like 

```
GET /up123/json HTTP/1.1
Rate-Limits: subscriber
```

Reply:
2: Good idea. The ntfy app subscribes to all the topics in one request, so it can't be a global header. Since we only want some topics to be subscriber-based-rate-limited, an alternative could be making a `Subscriber-Rate-Limit-Topics` header?

a. In `parseSubscribeParam` get a list of Subscriber-Rate-Limit-Topics.
b. In topic.Subscribe, pass whether that topic is in the list.
c. The topic can handle getBillee (or whatever that is renamed to accordingly)

Does that sound good?

Reply:
Yup, that sounds perfect, except for the really long header name. But we can fix that later. I do think that `up*` topics can temporarily be automatically subscriber-rate-limited, so maybe build that in, too?